processing/src/main/java/org/apache/druid/math/expr/vector/VectorProcessors.java

```diff
@@ -85,11 +85,14 @@ public static <T> ExprVectorProcessor<T> makeSymmetricalProcessor(
    *
    * @see org.apache.druid.math.expr.ConstantExpr
    */
-  public static <T> ExprVectorProcessor<T> constant(@Nullable String constant, int maxVectorSize)
+  public static <T> ExprVectorProcessor<T> constant(@Nullable Object constant, int maxVectorSize, ExpressionType type)
   {
-    final Object[] strings = new Object[maxVectorSize];
-    Arrays.fill(strings, constant);
-    final ExprEvalObjectVector eval = new ExprEvalObjectVector(strings, ExpressionType.STRING);
+    final Object[] objects = new Object[maxVectorSize];
+    if (type.isNumeric()) {
+      return constant((Long) null, maxVectorSize);
```

Comment:
why does this use `null` instead of looking at what `constant` is?

Reply:
oops, idk why this is here, i think this is from a bad merge between some stashes with experiments i had going on. It should just be removed, and wasn't failing any tests because numeric types use their dedicated constant functions